app/utility/base_planning_svc.py

```diff
@@ -98,7 +98,7 @@ async def remove_links_missing_facts(links):
         :return: updated list of links
         """
         links[:] = [l for l in links if
-                    not re.findall(r'#{(.*?)}', b64decode(l.command).decode('utf-8'), flags=re.DOTALL)]
+                    not re.findall(r'(#{[a-zA-Z1-9]+?\..+?})', b64decode(l.command).decode('utf-8'), flags=re.DOTALL)]
```

Comment:
I'm responsible for this change, so I thought I'd explain what's happening here. During testing, it became apparent that  remove_links_missing_facts is removing global variables because the regex it was using couldn't tell the different between them. This new regex only triggers on facts, not variables. I believe this change shouldn't break anything (if anything, it should restore functionality that I believe was technically broken), and it has been tested with Alice on multiple planners within AWS.